packages/nymjs/src/prover.ts

```diff
@@ -0,0 +1,70 @@
+// NOTE: this file should be a reference for frontend code
+import { hashPersonalMessage, fromRpcSig } from "@ethereumjs/util";
+
+import { computeEffEcdsaPubInput } from "@personaelabs/spartan-ecdsa";
+
+import { buildPoseidon } from "circomlibjs";
+
+import { MerkleProof, NymInput, EffECDSASig } from "./types";
+
+function bufferToBigInt(buf: Buffer): bigint {
```

Comment:
@lsankar4033 will we eventually upstream these functions into one package? I'm seeing their re-use in `circuits/test/test_utils.ts`

Reply:
Cool. I like duplicating functions as needed until we see where things can be re-used + upstreamed.